homeassistant/components/shelly/light.py

```diff
@@ -369,3 +408,27 @@ def _update_callback(self) -> None:
         self.control_result = None
         self.mode_result = None
         super()._update_callback()
+
+
+class RpcShellyLight(ShellyRpcEntity, LightEntity):
+    """Entity that controls a light on RPC based Shelly devices."""
+
+    def __init__(self, wrapper: RpcDeviceWrapper, key: str) -> None:
+        """Initialize light."""
+        super().__init__(wrapper, key)
+        self._id = key[7:]
```

Comment:
Seems fragile.

Reply:
The key is generated here:
https://github.com/home-assistant/core/blob/0b7cc6beb094880ad89c806d0d4cc0e95bf92deb/homeassistant/components/shelly/light.py#L112
The reason I need to create an `id` from it is that while for reading the switch state I need to get the state from `switch:id` for setting I need to call a method with the parameter `{"id": id, ...}`.
I could pass them both to the entity but preferred to use the key as identifier for both.

Reply:
Can we pass in the `id` to the constructor ?

Reply:
Yes